src/controls/DragControls.ts

```diff
@@ -0,0 +1,266 @@
+import { Camera, EventDispatcher, Intersection, Matrix4, Object3D, Plane, Raycaster, Vector2, Vector3 } from 'three'
+
+class DragControls extends EventDispatcher {
+  enabled: boolean
+  transformGroup: boolean
+
+  private _objects: Object3D[]
+  private _camera: Camera
+  private _domElement: HTMLElement
+
+  private _plane: Plane
+  private _raycaster: Raycaster
+
+  private _mouse: Vector2
+  private _offset: Vector3
+  private _intersection: Vector3
+  private _worldPosition: Vector3
+  private _inverseMatrix: Matrix4
+  private _intersections: Intersection[]
+
+  private _selected: Object3D | null
+  private _hovered: Object3D | null
+
+  constructor(_objects: Object3D[], _camera: Camera, _domElement: HTMLElement) {
+    super()
+
+    this._objects = _objects
+    this._camera = _camera
+    this._domElement = _domElement
+
+    this._plane = new Plane()
```

Comment:
i think this can come later. we will never finish otherwise, again, 382 files. that's just how they made it and maybe we can amend little by little. but really "fixing" these files, a real class with class methods, privates and publics - this would have to be done for everything and i see no way that can happen. if it runs and is usable we should move on until this project establishes itself.